packages/react/src/Image/Image.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import clsx from 'clsx'
+import styles from './Image.module.css'
+import {BaseProps} from '../component-helpers'
+
+type AspectRatio = '1:1' | '16:9' | '16:10' | '4:3' | 'custom'
+
+export type ImageProps = React.ImgHTMLAttributes<HTMLImageElement> &
+  BaseProps<HTMLImageElement> & {
+    src: string
+    alt: string
+    aspectRatio?: AspectRatio
+    media?: string
+  } & (
+    | {
+        as?: 'img' // assuming this is true for others to be required
+        srcSet?: 'string'
+      }
+    | {
+        as: 'picture' // assuming this is true for others to be required
+        sources?: {
+          srcset: string
+          media: string
+        }[]
+      }
+  )
+
+const aspectRatioResolver = (ratio?: AspectRatio) => {
+  if (typeof ratio === 'string') {
+    if (ratio === 'custom') return 'custom'
+    const [width, height] = ratio.split(':').map(Number)
+    return `${width}-${height}`
+  }
+}
+
+export const Image = ({
+  className,
+  aspectRatio,
+  as = 'img',
+  ref,
+  alt,
+  width,
+  height,
+  media,
+  srcSet,
+  ...rest
+}: ImageProps) => {
+  if (as === 'picture') {
+    return (
+      <picture
+        className={clsx(
+          styles['Image__container'],
+          aspectRatio && styles[`Image--aspect-ratio-${aspectRatioResolver(aspectRatio)}`]
+        )}
+      >
+        {rest['sources'] &&
+          rest['sources'].map((source, index) => <source key={index} srcSet={source.srcset} media={source.media} />)}
```

Comment:
When we're reading `sources` like this, should we pull this out of `rest` so that it doesn't get spread onto `<img>` below? Otherwise I think we end up with it stringifying into `sources="[object Object],[objectObject]"` like in the docs example

![image](https://user-images.githubusercontent.com/3901764/233469920-44f6734e-872b-418c-a0f0-dca04264230e.png)